java/src/main/java/cucumber/api/java/ObjectFactory.java

```diff
@@ -2,34 +2,10 @@
 
 /**
  * Minimal facade for Dependency Injection containers
+ *
+ * @deprecated use @code io.cucumber.core.backend.ObjectFactory instead
  */
-public interface ObjectFactory {
+@Deprecated
+public interface ObjectFactory extends io.cucumber.core.backend.ObjectFactory {
```

Comment:
Given that you've added an `ObjectFactoryAdapter` is it necessary to extend this interface?

Reply:
Not sure, but if not we have in both interfaces the same methods? And we need this Adapter (which I've only created not to change JavaBackend in a incompatible way?)
Let me know which solution you prefer:
1. Old Interface empty (current solution)
2. Old and new Interface are identical and only in different packages

Reply:
The duplicate code is not a problem. They're different interfaces. So option #2 would be fine.